extensions/eclipse-che-theia-plugin-ext/src/common/converter.ts

```diff
@@ -0,0 +1,52 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import * as che from '@eclipse-che/plugin';
+import * as theia from '@theia/task/lib/common';
+
```

Comment:
If I understand correctly we have these functions because `scope`field only in task config.
Can we align the corresponding object to upstream and avoid adding/using these functions?

Reply:
No, I don't think so: I'd have to start modifying the execution part of the tasks plugin, which I don't want to do at this time. Also, using theia extension types for our main/ext interface is not a good thing, IMO: main/ext usually uses their own types. This is more work, but makes the code less susceptible to changes in Theia. Otherwise, we'd have to use the same types everywhere (including the plugin side), which would break layering/modularity.